stdlib/public/Concurrency/AsyncIteratorProtocol.swift

```diff
@@ -12,9 +12,78 @@
 
 import Swift
 
+/// A type that that asychronously supplies the values of a sequence one at a
+/// time.
+///
+/// The `AsyncIteratorProtocol` defines the type returned by the
+/// `makeAsyncIterator()` method of the `AsyncSequence` protocol. In short,
+/// the iterator is what produces the asynchronous sequence's values. The
+/// protocol defines a single asynchronous method, `next()`, which either
+/// produces the next element of the sequence, or returns `nil` to signal
+/// the end of the sequence.
+///
+/// To implement your own `AsyncSequence`, implement a wrapped type that
+/// conforms to `AsyncIteratorProtocol`. The following example shows a `Counter`
+/// type that uses an inner iterator to monotonically generate `Int` values
+/// until a `howHigh` value is reached. While this example is not itself
+/// asychronous, it shows the shape of a custom sequence and iterator, and can
+/// be used as if it were asynchronous:
+///
+///     struct Counter : AsyncSequence {
+///       typealias Element = Int
+///       let howHigh: Int
+///
+///       struct AsyncIterator : AsyncIteratorProtocol {
+///         let howHigh: Int
+///         var current = 1
+///         mutating func next() async -> Int? {
+///           // A genuinely asychronous implementation could use the `Task`
```

Comment:
I think that perhaps call out `isCancelled` and `withTaskCancellationHandler` might be good? and I would claim it _should_ (implying obligation) rather than *could* (implying an option that is reasonable). All well behaved AsyncSequence adopters should handle cancellation in some form; either by checking the immediate `isCancelled` or the `checkCancellation` or `withTaskCancellationHandler`.